akka-actor/src/main/scala/akka/io/dns/DnsResourceRecords.scala

```diff
@@ -8,36 +8,39 @@ import java.net.{ Inet4Address, Inet6Address, InetAddress }
 
 import akka.actor.NoSerializationVerificationNeeded
 import akka.annotation.{ ApiMayChange, InternalApi }
+import akka.io
+import akka.io.FiniteCache
 import akka.io.dns.internal.{ DomainName, _ }
-import akka.util.{ ByteIterator, ByteString, ByteStringBuilder }
+import akka.util.{ ByteIterator, ByteString }
 
 import scala.annotation.switch
+import scala.concurrent.duration._
 
 @ApiMayChange
-sealed abstract class ResourceRecord(val name: String, val ttlInSeconds: Int, val recType: Short, val recClass: Short)
+sealed abstract class ResourceRecord(val name: String, val ttl: FiniteCache, val recType: Short, val recClass: Short)
```

Comment:
DNS RFC states the TTL is a positive 32 bit integer so the appropriate type here is `FiniteCache` since `NeverCache` and `CacheForever` are forbidden.

Reply:
BTW, here's the link: https://www.ietf.org/rfc/rfc1035.txt